### PR TITLE
ui5 Forms

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: src/shared/ResourceForm/fields/TextArrayInput.js [hunk 1]
```diff
@@ -34,28 +36,28 @@ export function TextArrayInput({
...
+            className="input-full"
+            // onKeyDown={e => focus(e)}
+            // onBlur={() => {
+            //   const fieldValue = internalValue?.filter(val => !!val);
+            //   setMultiValue(
+            //     typeof customFormatFn === 'function'
+            //       ? customFormatFn(fieldValue)
+            //       : fieldValue,
+            //   );
+            // }}
```
Comment (mrCherry97) [inline src/shared/ResourceForm/fields/TextArrayInput.js:57]: this is not adjusted as I see it properly, we need custom functionality for this

File: src/shared/ResourceForm/inputs/Number.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import React from 'react';
-import { FormInput } from 'fundamental-react';
+import { Input } from '@ui5/webcomponents-react';
 
 export function Number({ value = '', setValue, ...props }) {
+  if (!props.readOnly) delete props.readOnly;
```
Comment (mrCherry97) [inline src/shared/ResourceForm/inputs/Number.js:5]: I don't get it, is it really necessary?

File: src/shared/components/GenericList/SearchInput.js [hunk 3]
```diff
@@ -101,14 +106,13 @@ export function SearchInput({
...
-      const inputField = searchInputRef.current;
-      inputField.focus();
+      searchInput.focus();
     });
   };
 
   const handleOnKeyDown = e => {
-    const ESCAPE_KEY_CODE = 27;
-    if (e.keyCode === ESCAPE_KEY_CODE) {
+    console.log('handleOnKeyDown', e);
```
Comment (mrCherry97) [inline src/shared/components/GenericList/SearchInput.js:114]: clg

File: src/shared/components/GenericList/SearchInput.js [hunk 4]
```diff
@@ -130,17 +135,16 @@ export function SearchInput({
...
-            <input
+            <Input
+              id="search-input"
               aria-label="search-input"
-              ref={searchInputRef}
-              type="search"
-              placeholder={t('common.tooltips.search')}
-              value={searchQuery}
               onBlur={() => setSearchHidden(true)}
               onFocus={() => setSearchHidden(false)}
```
Comment (mrCherry97) [inline src/shared/components/GenericList/SearchInput.js:142]: this 2 props are not existing in UI5, so probably they are doing nothing

File: src/shared/components/K8sNameInput/K8sNameInput.js [hunk 5]
```diff
@@ -28,21 +29,21 @@ export const K8sNameInput = ({
...
+  const inputClassName = classnames('input-full', className, {
+    'ui5-content-density-compact': compact,
   });
 
   const input = (
-    <FormInput
-      ref={_ref}
-      type="text"
+    <Input
+      // ref={_ref}
```
Comment (mrCherry97) [inline src/shared/components/K8sNameInput/K8sNameInput.js:41]: ref doesn't exist in UI5, so we can potentially remove all of the refs for input components

File: src/shared/components/MemoryInput.js [hunk 6]
```diff
@@ -19,13 +19,12 @@ export function MemoryInput({ label, propertyPath, value = '', setValue }) {
 
   return (
     <div className="memory-input">
-      <FormInput
-        compact
+      <Input
         type="number"
```
Comment (mrCherry97) [inline src/shared/components/MemoryInput.js:23]:
```suggestion
        type="Number"
```

**Review by mrCherry97 — COMMENTED** [review]
I can only comment, because I was opening a PR :D

File: src/shared/ResourceForm/inputs/Number.js (hunk 2 above)
Comment (akucharska) [reply src/shared/ResourceForm/inputs/Number.js:5]: Yup, if readOnly is present (even if it is set to false) then input is in readOnly mode

File: src/shared/ResourceForm/fields/TextArrayInput.js (hunk 1 above)
Comment (akucharska) [reply src/shared/ResourceForm/fields/TextArrayInput.js:57]: We might have to adjust it later once this functionality is added to UI5 webcomonents. The workaround is very heavy on memory and the page gets into infinite loop if we try to add eventListener in array

File: src/components/Extensibility/components-form/SimpleList.js [hunk 7]
```diff
@@ -113,33 +115,35 @@ export function SimpleList({
...
+                        </div>
+                        <div className="fd-col-md--1">
+                          <span className="item-action">
+                            {!isLast(index) && (
+                              <Button
+                                disabled={readOnly}
+                                icon="delete"
+                                design="Negative"
+                                onClick={() => removeItem(index)}
+                                ariaLabel={t('common.buttons.delete')}
```
Comment (mrCherry97) [inline src/components/Extensibility/components-form/SimpleList.js:141]: change this to `aria-label`

File: tests/integration/tests/cluster/test-cluster-role-bindings.spec.js [hunk 8]
```diff
@@ -31,10 +31,10 @@ context('Test Cluster Role Bindings', () => {
 
     cy.contains('ui5-li:visible', 'cluster-admin').click();
 
-    cy.get('[ariaLabel="User name"]')
-      .click()
-      .clear()
-      .type(USER_NAME);
+    cy.get('[arialabel="User name"]')
```
Comment (mrCherry97) [inline tests/integration/tests/cluster/test-cluster-role-bindings.spec.js:34]: change this to `aria-label`

File: tests/integration/tests/cluster/test-cluster-role-bindings.spec.js [hunk 9]
```diff
@@ -67,10 +67,10 @@ context('Test Cluster Role Bindings', () => {
 
     cy.contains('Group').click();
 
-    cy.get('[ariaLabel="Group name"]')
-      .click()
-      .clear()
-      .type('test-group');
+    cy.get('[arialabel="Group name"]')
```
Comment (mrCherry97) [inline tests/integration/tests/cluster/test-cluster-role-bindings.spec.js:70]: change this to `aria-label`

**Review by mrCherry97 — COMMENTED** [review]
finally, we need to use `aria-label` instead of ariaLabel or arialabel, because we are getting errors in the console.

**Review by mrCherry97 — COMMENTED** [review]
Something is not yes
image

image

**Review by mrCherry97 — COMMENTED** [review]
LGTM

**Pull request merged**